bson/json_util.py

```diff
@@ -666,15 +665,16 @@ def _parse_canonical_regex(doc):
     if len(regex) != 2:
         raise TypeError('Bad $regularExpression must include only "pattern"'
                         'and "options" components: %s' % (doc,))
-    return Regex(regex['pattern'], regex['options'])
+
+    # Validate options, defer to Regex for pattern validation.
+    options = regex['options']
+    if not isinstance(options, str):
+        raise TypeError("pattern must be a string, not %s" % type(options))
+    return Regex(regex['pattern'], options)
```

Comment:
Ignore this, I'll rebase on top of https://github.com/mongodb/mongo-python-driver/pull/721